wiki/wiki/doctype/wiki_page/wiki_page.py

```diff
@@ -50,17 +50,24 @@ def on_update(self):
 		update_index(self)
 
 	def on_trash(self):
-		frappe.db.sql("DELETE FROM `tabWiki Page Revision Item` WHERE wiki_page = %s", self.name)
-
-		frappe.db.sql(
-			"""DELETE FROM `tabWiki Page Revision` WHERE name in
-			(
-				SELECT name FROM `tabWiki Page Revision`
-				EXCEPT
-				SELECT DISTINCT parent from `tabWiki Page Revision Item`
-			)"""
+		if frappe.db.exists('Wiki Page Revision Item', {'wiki_page': self.name}):
+			frappe.db.delete('Wiki Page Revision Item', {'wiki_page': self.name})
```

Comment:
`frappe.db.delete_if_exists(...)`

Reply:
Hello @blaggacao, I am not getting you why are you trying to make it complex? I checked but i have not found **delete_if_exists** method in develop branch nor in version-15 branch. 
Please correct me if I'm wrong!!
Thank you!!

Reply:
https://github.com/frappe/frappe/blob/develop/frappe%2F__init__.py

Reply:
This is done!! Thank you @blaggacao

Reply:
https://github.com/frappe/semgrep-rules/pull/29